src/screens/ProfileScreen/elements/BioAndDMSetting.tsx

```diff
@@ -28,18 +29,33 @@ const CheckBoxCustom = (props: {value: boolean; label: string; disabled?: boolea
   return (
     <View style={styles.checkboxItem}>
       <View style={styles.checkboxItemContent}>
-        <CheckBox
-          disabled={props.disabled || false}
-          value={props.value}
-          onCheckColor={COLORS.gray110}
-          tintColors={{true: COLORS.signed_primary, false: COLORS.gray110}}
-          tintColor={COLORS.signed_primary}
-          onTintColor={COLORS.signed_primary}
-          onFillColor={COLORS.signed_primary}
-          style={styles.checkbox}
-          aria-checked={props.value}
-          onValueChange={() => {}}
-        />
+        {Platform.OS === 'android' ? (
+          <>
+            {props.value ? (
+              <IconCheck
+                width={dimen.normalizeDimen(16)}
+                height={dimen.normalizeDimen(16)}
+                fill={COLORS.signed_primary}
+                fillIcon={COLORS.gray110}
+              />
+            ) : (
+              <View style={styles.checkboxFalse} />
+            )}
+          </>
+        ) : (
+          <CheckBox
+            disabled={props.disabled || false}
+            value={props.value}
+            onCheckColor={COLORS.gray110}
+            tintColors={{true: COLORS.signed_primary, false: COLORS.gray110}}
+            tintColor={COLORS.signed_primary}
+            onTintColor={COLORS.signed_primary}
+            onFillColor={COLORS.signed_primary}
+            style={styles.checkbox}
+            aria-checked={props.value}
+            onValueChange={() => {}}
+          />
+        )}
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The `onValueChange` prop for the `CheckBox` component is currently an empty function. This might lead to unexpected behavior as it doesn't update the state when the checkbox value changes. If this is intentional and you don't want to handle the change event, please ignore this comment. Otherwise, consider passing a function that handles the checkbox value change.
```diff
- onValueChange={() => {}}
+ onValueChange={handleCheckboxChange}
```
Please replace `handleCheckboxChange` with your actual function name.



<!-- This is an auto-generated comment by OSS CodeRabbit -->